### PR TITLE
Static resource tables

### PR DESCRIPTION
**Pull request by barankyle**
## Summary

_A summary of changes being made in this PR_

## References

closes #_insert number here_

## Checklist
- [ ] If this PR is still a WIP, convert to a draft
- [ ] [ensure all checks pass](https://github.com/XRFoundation/XREngine/wiki/Testing-&-Contributing)
- [ ] When this PR is ready, mark it as "Ready for review"
- [ ] Changes have been manually QA'd
- [ ] Changes reviewed by at least 2 approved reviewer

## QA Steps

_List any additional steps required to QA the changes of this PR, as well as any supplemental images or videos._

### CODE REVIEW

File: packages/engine/src/scene/components/TypeComponent.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import { hookstate, none } from '@xrengine/hyperflux'
+
+import { Entity } from '../../ecs/classes/Entity'
+import {defineComponent, getComponent} from '../../ecs/functions/ComponentFunctions'
+import {NameComponent} from "./NameComponent";
+
+export const TypeComponent = defineComponent({
+  name: 'TypeComponent',
+
+  onInit: () => '',
```
Comment (HexaField) [inline packages/engine/src/scene/components/TypeComponent.ts:10]: should return true

**Review by HexaField — CHANGES_REQUESTED** [review]

File: packages/common/src/dbmodels/Image.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export interface ImageInterface {
+  id: string
+  name?: string
+  tags?: string[]
+  height: number
+  width: number
+  jpegStaticResourceId?: string
```
Comment (speigg) [inline packages/common/src/dbmodels/Image.ts:7]: this should just be `staticResourceId`

File: packages/common/src/interfaces/AudioInterface.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+export interface AudioInterface {
+  id: string
+  name?: string
+  tags?: string[]
+  duration: number
+  mp3StaticResourceId?: string
+  oggStaticResourceId?: string
+  mpegStaticResourceId?: string
```
Comment (speigg) [inline packages/common/src/interfaces/AudioInterface.ts:8]: No reason for these to be separate static resources. An item in the audio table should point to a single static resource. That resource can be represented in a variety of formats / lods.

File: packages/common/src/interfaces/ImageInterface.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  width: number
+  src: string
+  jpegStaticResourceId?: string
+  jpegStaticResource?: StaticResourceInterface
+  ktx2StaticResourceId?: string
+  ktx2StaticResource?: StaticResourceInterface
+  gifStaticResourceId?: string
+  gifStaticResource?: StaticResourceInterface
+  pngStaticResourceId?: string
+  pngStaticResource?: StaticResourceInterface
```
Comment (speigg) [inline packages/common/src/interfaces/ImageInterface.ts:17]: Likewise here; each item in the image table should point to a single static resource.

File: packages/common/src/interfaces/VideoInterface.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  name?: string
+  tags?: string[]
+  height: number
+  width: number
+  duration: number
+  mp4StaticResourceId?: string
+  m3u8StaticResourceId?: string
+  mp4StaticResource?: StaticResourceInterface
+  m3u8StaticResource?: StaticResourceInterface
+  thumbnail?: string
```
Comment (speigg) [inline packages/common/src/interfaces/VideoInterface.ts:14]: Same; each video should point to a single static resource

File: packages/common/src/interfaces/VolumetricInterface.ts [hunk 6]
```diff
@@ -0,0 +1,11 @@
+export interface VolumetricInterface {
+  id: string
+  name?: string
+  tags?: string[]
+  duration: number
+  src: string
+  thumbnail?: string
```
Comment (speigg) [inline packages/common/src/interfaces/VolumetricInterface.ts:7]: `thumbnailResourceId`

**Review by speigg — CHANGES_REQUESTED** [review]
So, generally, the idea is that whenever files are _tightly coupled_ to one another (e.g., uvol geometry data in .drcs and texture data in .mp4 ), or _interchangeable variants_ of one another (e.g., different encodings of the same content), they should be contained in a single resource. 

Each **Image** in the image table should point to a single image `resourceId`. 

Each **Volumetric** should contain a `thumbnailId` and a `resourceId`. 

Each **Model** should contain a `resourceId` (for it's model various files) a `thumbnailId` , and maybe later (if we find good workflows / use cases) additional resource ids for overriding materials or textures or whatnot. 

Each **Avatar** should contain a `resourceId`, a `thumbnailId`, a `modelId`, and possibly additional model id slots for accessories (hats, gloves, etc.). 

All of the other tables don't really have to be done in a single PR. It would be preferable to have a clean vertical slice that we can validate with an end-to-end workflow. 

Let's also make sure each static resource can be linked somehow to each project.

**Review by speigg — APPROVED** [review]

**Pull request merged**